go/obscuronode/enclave/enclave.go

```diff
@@ -867,6 +882,26 @@ func (e *enclaveImpl) encryptTxReceiptWithViewingKey(address common.Address, txR
 	return e.rpcEncryptionManager.EncryptWithViewingKey(address, txReceiptBytes)
 }
 
+func (e *enclaveImpl) checkRollup(r *obscurocore.Rollup) {
+	stateDB := e.storage.CreateStateDB(r.Header.ParentHash)
+	// calculate the state to compare with what is in the Rollup
+	depositTxs := extractDeposits(
+		e.blockResolver.Proof(e.storage.ParentRollup(r)),
+		e.blockResolver.Proof(r),
+		e.bridge, e.blockResolver, e.erc20ContractLib, stateDB, e.config.ObscuroChainID,
+	)
+
+	rootHash, txReceipts, depositReceipts := e.process(r, stateDB, depositTxs)
+	// dump := stateDB.Dump(&state.DumpConfig{})
+	dump := ""
```

Comment:
Is this right?

Reply:
yeah, this is what will help us debug those "Calculated a different state" issues
But if we print them al time, it's a bit slow and heavy.